Sources/MessagingPush/RichPush/RichPushDeliveryTracker.swift

```diff
@@ -0,0 +1,43 @@
+import CioInternalCommon
+import Foundation
+
+// sourcery: InjectRegisterShared = "RichPushDeliveryTracker"
+class RichPushDeliveryTracker {
+    let httpClient: HttpClient
+    let logger: Logger
+
+    init(httpClient: HttpClient, logger: Logger) {
+        self.httpClient = httpClient
+        self.logger = logger
+    }
+
+    func trackMetric(token: String, event: Metric, deliveryId: String, timestamp: String? = nil, onComplete: @escaping (Result<Void, HttpRequestError>) -> Void) {
+        let properties: [String: Any] = [
+            "anonymousId": deliveryId,
+            "properties": [
+                "recipient": token,
+                "metric": event.rawValue,
+                "deliveryId": deliveryId
+            ],
+            "event": "Report Delivery Event"
+        ]
+
+        let endpoint: CIOApiEndpoint = .trackNSEPushDeliveryMetrics
+        guard let httpParams = HttpRequestParams(
+            endpoint: endpoint,
+            baseUrl: RichPushHttpClient.defaultAPIHost,
+            headers: nil,
+            body: try? JSONSerialization.data(withJSONObject: properties)
```

Comment:
Is there a reason for not using `jsonAdapter` here?

Reply:
I think I tried answering this in previous PRs too, so let me explain with an example. `jsonAdapter` is not that agnostic is meant to work as an adapter for json that our `tracking API` expected and accepted. 

For example in this case, if were to use the method of `jsonAdapter` 

```
 public func fromDictionary<T: Decodable>(_ dictionary: [AnyHashable: Any]) -> T? {
        do {
            let jsonData = try JSONSerialization.data(withJSONObject: dictionary)

            return fromJson(jsonData)
        } catch {
            log.error("\(error.localizedDescription), dictionary: \(dictionary)")
        }

        return nil
    }
```

`let jsonData = try JSONSerialization.data(withJSONObject: dictionary)` it's great and solves our problem, but later you see the very next line, `fromJson(jsonData)`, it starts decoding this `json` and breaks it. 

Now I can go ahead and start creating these new methods in `jsonAdapter` but again `jsonAdapter` is biased, you see the decode being used, now that wouldn't work too because to be compatible with Segment we need to use it `iso8601` as date decoding strategy. 

```
var decoder: JSONDecoder {
        let decoder = JSONDecoder()
        decoder.dateDecodingStrategy = .secondsSince1970
        return decoder
    }
```


Hence, I avoided using the `jsonAdapter` too aggressively or any other class for the utility that's not agnostic, for cleanup later on we can create the utility class but currently, the usage is too low to need one imho.

Reply:
If `JsonAdapter` is not being used in multiple places, then I am OK with not making a change in this current PR. 


The implementation of `JsonAdapter` today could be bias towards Track.